src/test/scala/com/vngrs/etl/BaseSparkSpec.scala

```diff
@@ -0,0 +1,39 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.BeforeAndAfterAll
+
+import scala.reflect.ClassTag
+import scala.util.Random
+
+/**
+  * Base Specification class which handles SparkContext creation and demolition
+  */
+abstract class BaseSparkSpec extends BaseSpec with BeforeAndAfterAll {
+  private val conf = new SparkConf().setMaster("local[*]").setAppName(generateAppName())
+  // TODO make SparkContext private but also provide a way to supply it to Pipeline implicitly
+  protected implicit val sc: SparkContext = new SparkContext(conf)
```

Comment:
@msimav what do you think about this? I had to make `protected implicit` it for testing `Pipeline` factory methods.